tests/ModelParser/JMSParserTest.php

```diff
@@ -167,6 +167,42 @@ public function providePropertyType(): iterable
             true,
             'string[]|null',
         ];
+
+        yield [
+            new class() {
+                /**
+                 * @JMS\Type("ArrayCollection<string>")
+                 */
+                private $property;
+            },
+            PropertyTypeArray::class,
+            true,
+            'string[]|\Doctrine\Common\Collections\Collection|null',
+        ];
+
+        yield [
+            new class() {
+                /**
+                 * @JMS\Type("Doctrine\Common\Collections\ArrayCollection<string>")
+                 */
+                private $property;
+            },
+            PropertyTypeArray::class,
+            true,
+            'string[]|\Doctrine\Common\Collections\Collection|null',
+        ];
+
+        yield [
+            new class() {
+                /**
+                 * @JMS\Type("ArrayCollection<string, int>")
+                 */
+                private $property;
+            },
+            PropertyTypeArray::class,
+            true,
+            'int[string]|\Doctrine\Common\Collections\Collection|null',
```

Comment:
could this specify key and value, with `...\Collection<int, string>...`?

Reply:
I adjusted the `__toString()` method via a95fd9aec4fb795ac183a9760b8464324ac4e400